git_machete/client.py

```diff
@@ -1674,16 +1674,22 @@ def checkout_github_prs(self, pr_no: int) -> None:
         repo: str
         remote: str
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        current_user: Optional[str] = git_machete.github.derive_current_user_login()
         debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
         print(f"Fetching {remote}...")
         self.__git.fetch_remote(remote)
-        self.flush_caches()
 
         pr = get_pull_request_by_number_or_none(pr_no, org, repo)
         if not pr:
             raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
         if '/'.join([remote, pr.head]) not in self.__git.get_remote_branches():
-            raise MacheteException(f"Could not check out PR #{pr_no} because its head branch `{pr.head}` is already deleted from `{remote}`.")
+            remote_from_pr: str = pr.full_repository_name.split('/')[0]
+            if remote_from_pr not in self.__git.get_remotes():
+                self.__git.add_remote(remote_from_pr, pr.repository_url)
+            print(f"Fetching {remote_from_pr}...")
+            self.__git.fetch_remote(remote_from_pr)
+            if '/'.join([remote_from_pr, pr.head]) not in self.__git.get_remote_branches():
+                raise MacheteException(f"Could not check out PR #{pr_no} because its head branch `{pr.head}` is already deleted from `{remote_from_pr}`.")
```

Comment:
Huh another unexpected behavior 🐞 

```
github checkout-prs 204
Fetching origin...

Could not check out PR #204 because its head branch remove/spam-displayed is already deleted from origin.
```

But... `remove/spam-displayed`/PR #204 do not correspond to `origin`, rather to `garg-tanishka` fork

Reply:
For clarity, I don't have local `remove/spam-displayed` branch but I do have `garg-tanishka/remove/spam-displayed`

Reply:
Opened as #254 for easier tracking